packages/frontend/src/common/dialogs/ParticipantsDialog.tsx

```diff
@@ -37,8 +44,19 @@ export function ParticipantsDialog({
 }: ParticipantDialogProps) {
   const theme = useTheme();
   const fullScreen = useMediaQuery(theme.breakpoints.down("sm"));
-
+  const { isAutoAcceptActivated, setIsAutoAcceptActivated, roomId } = useRoomContext();
+  const namespace = useNamespace();
   const isDividerVisible = !isEmpty(waitingList) && !isEmpty(participants);
+  const { user } = useUserContext();
+  const [isSwitchDisabled, setIsSwitchDisabled] = useState(false);
+
+  async function toggleChecked() {
+    setIsSwitchDisabled(true);
+    const toggledValue = !isAutoAcceptActivated;
+    setIsAutoAcceptActivated(toggledValue);
+    await putIsAutoAcceptActivated({ roomId, namespace, isActivated: toggledValue });
+    setIsSwitchDisabled(false);
+  }
```

Comment:
Why do we disable the switch first and enable it again after the request was fired?